Makefile

```diff
@@ -100,36 +125,43 @@ GENERATED_FILES_DEP = \
 	test/dune \
 
 generated: $(GENERATED_FILES_DEP)
+	@printf "Done.\n"
 
-install uninstall fmt build distrib: info $(GENERATED_FILES_DEP)
+fmt build gwd install uninstall: info patch_files generated
 
 fmt: ## Format Ocaml code
 ifneq ($(OS_TYPE),Win)
-	@printf "\n\033[1;37mOcamlformat\033[0m\n"
-	dune build @fmt --auto-promote
+	@printf "\n\033[1;1mOcamlformat\033[0m\n"
+	dune build @fmt --auto-promote ; $(UNPATCH)
 endif
 
 # [BEGIN] Installation / Distribution section
 
 build: ## Build the geneweb package (libraries and binaries)
-build:
-	@printf "\n\033[1;37mBuilding executables\033[0m\n"
-	dune build -p geneweb --profile $(DUNE_PROFILE)
+	@printf "\n\033[1;1mBuilding executables\033[0m\n"
+	@$(BUILD) ; $(UNPATCH)
```

Comment:
I believe build target will now succeed even if $(BUILD) fails. You probably want `&&` here.

Reply:
I know it works with the current setup!

Reply:
FYI, I'm MacPorts maintainer of geneweb and was made aware of this camlp5 issue by a user who installed geneweb on a platform where we hadn't a prebuilt binary after camlp5 was updated. See: http://trac.macports.org/ticket/70174

Concerning ; instead of &&, the logics in `build` and `distrib` targets now differ, while previously `distrib` target simply depended on `build` target. I noticed this because of a particular binary incompatibility on my machine, unrelated to this PR, and `distrib` target failed while `build` succeeded (but there actually was a build error).

Regarding `not-ocamlfind`, I'm not aware that `camlp5` depends on it. Indeed, I had `camlp5` 8.03.00 on my machine but not `not-ocamlfind`. And looking at opam file, it's not listed:
https://github.com/camlp5/camlp5/blob/master/opam
But maybe I lack some ocaml packaging knowledge here.

I fixed this by adding `not-ocamlfind` to MacPorts and geneweb now can be compiled again using this very PR's patch. See: https://github.com/macports/macports-ports/pull/24362 I confirm it works, and you can see that the CI passes on several versions of macOS.

Reply:
Yes, that caml5 problem is latent since 25 May 2024… I previously changed the `distrib` target to include `build` in a previous commit already in the master, that was to simplify builds. I reverted that behaviour here, because I couldn’t solve that new problem for the build target without it. I repeat I hope this is temporary, and it would be better to remove camlp5 totally. So this little change makes the `distrib` target autonomous as it was before. I’m checking again about ’not-ocamlfind’ and will answer again later.

PS: I have done a PR testing all Github Action CI available for all macOS version even macOS 11 that will be soon depreciated here, PR is from last month #1794 and nearly no change in master since that. And this more recent patch/unpatch thing has been tested on it for Ocaml 4.14 and 5.2 only atm + on a Colab=Ubuntu for Ocaml 4.09. No pb under Windows because it doesn’t have last camlp5 available. So I think it’s OK for all passing builds of the given table.

Reply:
@pguyot Asking opam to install only `camlp5` on a new Ocaml switch, I'm now quite confident that `not-ocamlfind` is installed from a “chain of dependancies” when installing its last version. See:

```ocaml
$opam install camlp5
[WARNING] Running as root is not recommended
The following actions will be performed:
  ∗ install conf-perl           2       [required by camlp5]
  ∗ install rresult             0.7.0   [required by camlp5]
  ∗ install conf-which          1       [required by not-ocamlfind]
  ∗ install conf-bash           1       [required by camlp5-buildscripts]
  ∗ install fmt                 0.9.0   [required by camlp5]
  ∗ install ocamlgraph          2.1.0   [required by not-ocamlfind]
  ∗ install astring             0.8.5   [required by bos]
  ∗ install conf-m4             1       [required by not-ocamlfind]
  ∗ install logs                0.7.0   [required by bos]
  ∗ install fpath               0.7.3   [required by bos]
  ∗ install not-ocamlfind       0.13    [required by camlp5-buildscripts]
  ∗ install bos                 0.2.1   [required by camlp5]
  ∗ install camlp5-buildscripts 0.03    [required by camlp5]
  ∗ install camlp5              8.03.00
===== ∗ 14 =====
Do you want to continue? [Y/n]
```

Reply:
@a2line you are right. `camlp5-buildscripts` is not strictly required to build camlp5 but declared as a dependency of it, and this is how you eventually have `not-ocamlfind` installed. I'll port the missing packages to MacPorts and fix the dependencies there.